webapp/src/js/services/live-list.js

```diff
@@ -431,55 +434,44 @@ angular.module('inboxServices').factory('LiveList',
       }
       if (removeIndex !== null) {
         idx.list.splice(removeIndex, 1);
-        var removed = idx.dom.splice(removeIndex, 1);
+        const removed = idx.dom[removedItem._id];
+        delete idx.dom[removedItem._id];
 
         $(idx.selector).children().eq(removeIndex).remove();
-        if (removed.length) {
-          return removed[0];
-        }
+        return removed;
       }
     }
 
     function _setSelected(listName, _id) {
-      var i, len, doc,
-          idx = indexes[listName],
-          list = idx.list,
-          previous = idx.selected;
+      const idx = indexes[listName],
+            previous = idx.selected;
 
       idx.selected = _id;
 
-      if (!list) {
+      if (!idx.list) {
         return;
       }
 
-      for (i=0, len=list.length; i<len; ++i) {
-        doc = list[i];
-        if (doc._id === previous) {
-          idx.dom[i]
-              .removeClass('selected');
-        }
-        if (doc._id === _id) {
-          idx.dom[i]
-              .addClass('selected');
-        }
+      if (idx.dom[_id]) {
+        idx.dom[_id].addClass('selected');
+      }
+      
+      if (previous && idx.dom[previous]) {
+        idx.dom[previous].removeClass('selected');
```

Comment:
Switching these two around (deselect previous then select new => select new then deselect previous) causes reports selected status to not show up properly in some cases. For example, select a report and refresh, the report will not appear as selected on RHS. Same thing happens when you add a new report (RHS shows the report but it is not selected in LHS).

Reply:
I can't reproduce the specific failure you mentioned -- but I totally see how it could happen. My intention was not to change anything with the refactor. I've changed back to the original order.